src/components/Search/SearchList.tsx

```diff
@@ -125,19 +133,23 @@ function SearchList(
 
     const {initialHeight, initialWidth} = useInitialWindowDimensions();
     const {hash, groupBy, type} = queryJSON;
-    const flattenedTransactions = groupBy ? (data as TransactionGroupListItemType[]).flatMap((item) => item.transactions) : data;
-
-    const flattenedTransactionWithoutPendingDelete = useMemo(
-        () => flattenedTransactions.filter((t) => t?.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE),
-        [flattenedTransactions],
-    );
+    const flattenedItems = useMemo(() => {
+        if (groupBy) {
+            if (!isTransactionGroupListItemArray(data)) {
+                return data;
+            }
+            return data.flatMap((item) => item.transactions);
+        }
+        return [];
+    }, [data, groupBy]);
+    const flattenedItemsWithoutPendingDelete = flattenedItems.filter((t) => t?.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE);
```

Comment:
```suggestion
    const flattenedItemsWithoutPendingDelete = useMemo(() => flattenedItems.filter((t) => t?.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE, [flattenedItems]);
```

This can be helpful if the search list items are large.